gapir/client/background_connection.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/app/crash/reporting"
+	"github.com/google/gapid/core/app/status"
+	"github.com/google/gapid/core/data/id"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/os/device"
+	"github.com/google/gapid/core/os/file"
+	"github.com/google/gapid/gapir"
+	"github.com/google/gapid/gapis/database"
+)
+
+// ReplayExecutor handles just the bits related to a
+// specific replay.
+type ReplayExecutor interface {
+	// HandlePostData handles the given post data message.
+	HandlePostData(context.Context, *gapir.PostData) error
+	// HandleNotification handles the given notification message.
+	HandleNotification(context.Context, *gapir.Notification) error
+	// HandleFinished is notified when the given replay is finished.
+	HandleFinished(context.Context, error) error
+
```

Comment:
Missing description comment

Reply:
@apbodnar can you please give context here so that I can update the comment